Library/Homebrew/dev-cmd/audit.rb

```diff
@@ -50,19 +50,33 @@
 require "missing_formula"
 require "digest"
 
+require_relative "../cli_parser"
+
 module Homebrew
   module_function
 
   def audit
     inject_dump_stats!(FormulaAuditor, /^audit_/) if ARGV.switch? "D"
     Homebrew.auditing = true
+    args, _unnamed_args = HomebrewCli.options do
+      option :strict, "strict", type: :boolean, default: false
+      option :online, "online", type: :boolean, default: false
+      option :new_formula, "new-formula", type: :boolean, default: false
+      option :fix, "fix", type: :boolean, default: false
+      option :display_cop_names, "display cop names", type: :boolean, default: false
+      option :display_filename, "display file names", type: :boolean, default: false
+      option :only, "only", type: :string
+      option :except, "except", type: :string
+      option :only_cops, "only cops", type: :string
+      option :except_cops, "except cops", type: :string
+      conflicts :only_cops, :except_cops, :strict, :only
+    end
 
     formula_count = 0
     problem_count = 0
-
-    new_formula = ARGV.include? "--new-formula"
-    strict = new_formula || ARGV.include?("--strict")
-    online = new_formula || ARGV.include?("--online")
+    new_formula = args.new_formula
```

Comment:
`args.new_formula?` for all of these reads a bit nicer.

Reply:
Figuring out a way to do this.